src/sb_logger.c

```diff
@@ -497,3 +514,35 @@ int oper_handler_done(void)
 
   return 0;
 }
+
+char *create_pct_string_intermediate(double* percentiles, double* results, size_t npercentiles){
```

Comment:
Key place I am looking for advice on how to handle memory management and string concatenation in this function and the function below it (`create_pct_string_cumulative`). Lots of duplicated code between these two functions.